artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/PagedReferenceImpl.java

```diff
@@ -53,6 +53,8 @@
 
    private Object protocolData;
 
+   private Long messageSize;
```

Comment:
Could this be primitive? and either -1 or 0 to denote empty or not set in logic, just objects = GC, and we're putting effort in to remove them on the hot path as much as possible to keep GC pressure as low as possible.

Reply:
Yep, I'll set this to -1 to mean not set.